java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java

```diff
@@ -642,9 +660,38 @@ private static void alignEnvironment(ContentEnvironment env, Channel baseChannel
                 .sorted((t1, t2) -> Boolean.compare(t1.getChannel().isBaseChannel(), t2.getChannel().isBaseChannel()))
                 .forEach(toRemove -> ContentProjectFactory.purgeTarget(toRemove));
 
-        // align the contents
-        newSrcTgtPairs.forEach(srcTgt ->
-                alignEnvironmentTarget(srcTgt.getLeft(), srcTgt.getRight(), filters, async, user));
+        // Strip modules metadata from the modular repositories if any modular filter is present
```

Comment:
Just a question: Is stripping the metadata from the "target" channel something that needs to be done (to overcome some errors), or is it just for clarity (cloned channels shouldn't be modular).

Reply:
I'd say it's for both. Since we're filtering out the unselected modules, the existing metadata is not valid and would confuse the clients. This is to prevent `dnf` from treating the channel as modular.